### ISSUE TITLE
Duplicate filenames in rclone/Google Drive

### ISSUE THREAD

**Issue by probablykasper** [kopia/kopia#3217]
My rclone (Google Drive) repository had a bunch of folders with the same name in it, where one of the folders contains a blob and the other is empty. This caused my full maintenance runs to fail. `contents verify` found 600 missing blob errors.

Ideally Kopia would handle these errors somehow, and hopefully reduce creation of duplicates somehow.

Rclone has a [`dedupe`](https://rclone.org/commands/rclone_dedupe/) command to remove duplicates.

The Google Drive API has a [`files.generateIds`](https://developers.google.com/drive/api/reference/rest/v3/files/generateIds) endpoint, maybe that could improve reliability somehow

**Comment by probablykasper** [comment]
Yes. I doubt I'm the only one with this issue, I've seen several other users report these missing blob errors

**Comment by lupusA** [comment]
I think this needs reproduction. Maybe it is related to rclone or the storage provider itself. 

Cheers,

**Comment by jkowalski** [comment]
@probablykasper  Can you paste the details from your investigations including contents of the google drive for missing files?

You had duplicates in the drive, can you post their file information including modification times?

Also if you can find any logs that mention the blob IDs in question, please post them also.

**Comment by probablykasper** [comment]
Unfortunately I don't have the repository anymore because I had to save space, sorry :/ I'll keep a close eye on it for when it happens next time

**Comment by probablykasper** [comment]
Tried to back up my an external drive I have and I once again got the error again.
> 05:26:24.973 error processing Projects/Shrive/Archive/2020: error reading directory: unable to open object: kbaf1b6fdc11c2ef115d689d05260e998: unexpected content error: error getting cached content: failed to get blob with ID q311e11b4cc287f576440f8127bba0422-s2d9e9466c4ed9da111f: BLOB not found

In this case there are two `q31` folders - one empty and one with the `1e1` folder.

On the Google Drive website, everything has the creation/modification date 2023-08-17 - It doesn't specify the time. If I download or sync them, they have these various timestamps:

| Path           | Downloaded          | Synced
| -              | -                   | -
| q31/ (empty)   | 2023-08-17 22:37 | 2023-08-17 04:41
| q31/           | 2023-08-18 07:38 | 2023-08-17 04:47 (Created)<br>2023-08-18 07:27 (Modified)
| q31/1e1/       | 2023-08-18 07:38 | 2023-08-17 04:47
| q31/1e1/blob.f | 2023-08-16 19:47 | 2023-08-17 04:47

For some reason the snapshot "group" says it takes up 0 bytes, while the snapshot inside says 1 TB. In reality, the folder being backed up is 806 GB.

<img width="989" alt="image" src="https://github.com/kopia/kopia/assets/11315492/a8c4d55b-c782-48d5-9864-806b220aefed">
<img width="994" alt="image" src="https://github.com/kopia/kopia/assets/11315492/51826236-3911-41b7-987f-2534646fd39f">

I'd be happy to provide logs, but they don't show up `~/Library/Logs/kopia`, not sure why? I can access task logs in KopiaUI though.
